ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java

```diff
@@ -336,6 +350,38 @@ private LogicalSchema rebuildWithoutPseudoAndKeyColsInValue(final int pseudoColu
     return new LogicalSchema(builder.build());
   }
 
+
+  /**
+   * Rebuilds schema with only certain pseudocolumns to materialize in an intermittent store
+   * @param pseudoColumnVersion indicates which set of pseudocolumns should be used
+   * @return the LogicalSchema created, with the corresponding pseudocolumns
+   */
+  private LogicalSchema rebuildWithPseudoColumnsToMaterialize(final int pseudoColumnVersion) {
+    final Map<Namespace, List<Column>> byNamespace = byNamespace();
+
+    final ImmutableList.Builder<Column> builder = ImmutableList.builder();
+
+    final List<Column> keyColumns = keyColumns(byNamespace);
+
+    final List<Column> nonPseudoAndKeyCols = nonPseudoAndKeyColsAsValueCols(
+        byNamespace, pseudoColumnVersion);
+
+    int valueIndex = nonPseudoAndKeyCols.size();
+
+    builder.addAll(keyColumns);
+    builder.addAll(nonPseudoAndKeyCols);
+
+    //put pseudocolumns which are not able to be accessed when getting value from an upstream
+    //state store, and therefore require materialization, in a conditional block corresponding to
```

Comment:
There is no need to describe "in a conditional block corresponding to its pseudocolumn version number" in the comment since the reader can see that from the code itself. In other words, comments are valuable if they make reading the code easier, but too many comments (especially redundant ones) can be harmful if they make reading the code harder :)

Reply:
I agree, this felt clunky when I wrote it but I wanted to be extra explicit that people don't add new pseudocolumns in the future that correspond with a different number into that block. Now that I'm thinking about it, I think that detail is already clear from how we've named `ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION`.